narwhal/primary/src/primary.rs

```diff
@@ -962,6 +979,23 @@ impl PrimaryToPrimary for PrimaryReceiverHandler {
                 .map_err(|e| anemo::rpc::Status::from_error(Box::new(e)))?
             {
                 Some(cert) => {
+                    if self.protocol_config.narwhal_certificate_v2() {
```

Comment:
Skipping serving a certificate may make subsequent certificates miss parents. We need to ensure all certificates served as fetch result can be verified directly or indirectly. And the optimization is to skip direct verifications when possible on the fetcher side.

Reply:
As discussed switching to serve ALL certificates in store and leave it up to the fetcher to verify.